fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -668,28 +657,29 @@ impl<'vm, S, I> CodeCopyCtx<'vm, S, I> {
         S: InterpreterStorage,
     {
         let contract = CheckedMemConstLen::<{ ContractId::LEN }>::new(b)?;
-        let cd = checked_add_word(c, d)?;
 
-        if d > MEM_MAX_ACCESS_SIZE
-            || a > checked_sub_word(VM_MAX_RAM, d)?
-            || cd > VM_MAX_RAM
-        {
-            return Err(PanicReason::MemoryOverflow.into())
-        }
+        MemoryRange::new(a, d)?;
```

Comment:
It would be nice, but it's possible to invalidate the checks before creating a range and using it, still imposing the responsibility to the caller. So, I think it's not that big of an improvement. https://github.com/FuelLabs/fuel-vm/pull/524 takes some minor steps to making things like this cleaner, and I'm hoping to continue to that direction in a future PR.

Reply:
`try_memory...` still does the full ownership checks. The reason we do the check here as well is to preserve the panic reason precedence. Not checking it here would make `ContractNotFound` trigger before the ownership checks. Maybe that's not an issue, but I didn't want to break any backwards compatibility needlessly.

Reply:
I don't know whether it matters which PanicReason is given as long as it's consistent, though, and the code certainly would be simpler if we didn't care.

Reply:
The order of panic checks shouldn't really matter much as long as we can unit test the code to verify each panic condition.